src/matter/interaction/InteractionMessenger.ts

```diff
@@ -190,7 +191,16 @@ export class InteractionServerMessenger extends InteractionMessenger<MatterDevic
             }
         }
 
-        await this.exchange.send(MessageType.ReportData, TlvDataReport.encode(dataReport));
+        if (dataReport.suppressResponse) {
+            // We do not expect a response other than a Standalone Ack, so if we receive anything else, we throw an error
+            await tryCatchAsync(async () => await this.exchange.send(MessageType.ReportData, TlvDataReport.encode(dataReport), true), UnexpectedMessageError, error => {
+                const { payloadHeader: { messageType }, payload } = (error as UnexpectedMessageError).receivedMessage;
```

Comment:
tryCatchAsync needs to correctly bind the expected type with the argument of the handler, so you should not have to class cast here.

Modify this:
type ErrorHandler<E extends Error, T> = (error: E) => T;

and

export async function tryCatchAsync<E extends Error, T>(codeBlock: () => Promise<T>, errorType: ClassExtends<E>, fallbackValueOrFunction: ErrorHandler<E, T> | T): Promise<T> {

Reply:
Cool, that solved a lot ... included now ...